### PR TITLE
Fixup AZP compilations and update to newer everything

### PR DESCRIPTION
**Pull request by jgunthorpe**
Various churn in Ubuntu and clang land broke building the azp image, fix it up, and update clang, gcc and base OSs.

Fedora should be updated too, but it is currently broken.

### CODE REVIEW

**Comment by rleon** [comment]
What is wrong with Fedora?

File: buildlib/cbuild [hunk 1]
```diff
@@ -491,6 +518,7 @@ environments = [centos6(),
                 amazonlinux2(),
                 xenial(),
                 bionic(),
+                focal(),
```
Comment (rleon) [inline buildlib/cbuild:521]: Keep it sorted?

File: buildlib/azure-pipelines.yml [hunk 2]
```diff
@@ -114,27 +117,27 @@ stages:
               set -e
               mkdir build-arm64
               cd build-arm64
-              CC=aarch64-linux-gnu-gcc-8 cmake -GNinja .. -DIOCTL_MODE=both -DNO_PYVERBS=1 -DENABLE_WERROR=1
+              CC=aarch64-linux-gnu-gcc-9 cmake -GNinja .. -DIOCTL_MODE=both -DNO_PYVERBS=1 -DENABLE_WERROR=1
```
Comment (rleon) [inline buildlib/azure-pipelines.yml:120]: Can you remind me why aren't we using latest gcc-10 for aarch64 and ppc?

**Comment by jgunthorpe** [comment]
Fedora's annobin is broken:

```
annobin: providers/mlx5/cq.c: ICE: attempting to access a gcc command line option that is not stored in global_options
annobin: providers/mlx5/cq.c: ICE: Please contact the annobin maintainer with details of this problem
annobin: providers/mlx5/cq.c: debugging: index = 853 max = 1919
```

Needs a new version which should come out someday:
https://bugzilla.redhat.com/show_bug.cgi?id=1817659

File: buildlib/azure-pipelines.yml (hunk 2 above)
Comment (jgunthorpe) [reply buildlib/azure-pipelines.yml:120]: ubuntu doesn't build the cross compilers for the gcc ppa

**Comment by rleon** [comment]
ok, let's wait till tomorrow and merge it then.

**Comment by galpress** [comment]
Verified this solves the azp issue. 👍

**Pull request merged**